app/models/desk.rb

```diff
@@ -2,4 +2,9 @@ class Desk < ApplicationRecord
   belongs_to :grouping, optional: true
 
   validates :name, presence: true
+
+  def relevant_changes
+    changed_keys = previous_changes.slice('name', 'occupied').keys
+    slice(changed_keys << 'id')
+  end
```

Comment:
Business logic: filters changes for those that a client cares about.

What do you think?

Reply:
Do you really need this method? Honestly if `broadcast_needed?` returns true I'd just broadcast the entire desk (or just the `name` and `occupied` attributes).

Reply:
This ensures minimal changes on the client, but then again I suppose this means the client has to check for what the message contains. If we always send the same message, it's easier on the client. I will change it.